webapp/src/types/model.ts

```diff
@@ -65,16 +65,17 @@ export type Project = {
     issuetypes: IssueType[];
 }
 
-export type IssueMetadata = {
-    projects: Project[];
-}
-
-export type ProjectStatuses = {
+export type IssueTypeWithStatuses = {
     id: string;
     name: string;
     statuses: Status[];
 }
 
+export type IssueMetadata = {
+    projects: Project[];
+    issue_types: IssueTypeWithStatuses[];
+}
```

Comment:
Is the sole purpose of having a separate `issue_types` here to have the statuses? Any way we can just have the backend update `project.issuetypes` to include the statuses? It seems like we may be returning twice the data in this code by duplicating the issue types. If we keep them separate then I think we should at least remove everything from `IssueTypeWithStatuses` that doesn't have to do with statuses before sending back to the client, if statuses is all it's being used for.

Edit: I think I understood this wrong. The `ListProjectStatuses` backend `JiraClient` function doesn't return all of the issue type data. I'll trust your judgment how this should be organized. I think we should be more specific with this name though like `issue_types_with_statuses`.

Reply:
@mickmister Yes, `ListProjectStatuses` returns only the fields that we need in our implementation. It would be very complex to move this inside the `project.issueTypes` as we have to manually iterate over both the array multiple times and add the field. Also, the type used for `issueTypes` inside the project data does not have the "statuses" field. So, we have to create a new custom type instead of using the type provided by the package.